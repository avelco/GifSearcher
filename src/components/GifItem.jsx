import React, { useState } from 'react';
import PropTypes from 'prop-types';

export default function GifItem({ title, url }) {
  const [showTooltip, setShowTooltip] = useState(false);

  const handleCopy = () => {
    navigator.clipboard.writeText(url);
    setShowTooltip(true);
    setTimeout(() => {
      setShowTooltip(false);
    }, 2000); // Hide the tooltip after 2 seconds
  };

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
      <button onClick={handleCopy}>Copy URL</button>
      {showTooltip && <div className="tooltip">URL Copied!</div>}
    </div>
  );
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};