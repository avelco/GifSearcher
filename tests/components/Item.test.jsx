import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Item", () => {
    const title = "Title";
    const url = "https://localhost/image.png";

    test("should render correctly", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

	test("Should show the images and the ALT text", () => {
		render(<GifItem title={title} url={url} />);

		// const img = document.querySelector("img");
		// expect(img.src).toBe(url);

		// const altText = img.alt;
		// expect(altText).toBe(title);

		const { alt, src } = document.querySelector("img");
		expect(alt).toBe(title);
		expect(src).toBe(url);
	});

	test("Should have the title in the paragraph", () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
