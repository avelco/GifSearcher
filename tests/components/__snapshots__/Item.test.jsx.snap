// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render correctly 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://localhost/image.png"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
